### PR TITLE
Static overview UI foundation

### PR DESCRIPTION
**Pull request by CHOOSEIT**
This pull request introduces foundational elements for the Overview UI. Fixes: [#30](https://github.com/ProximaEPFL/proxima/issues/30)

It includes two enums: `NavigationbarRoutes`, defining bottom navigation bar entries, and `TimeLineFilters`, specifying timeline filter options. 

Additionally, it introduces a sample provider named `postList` that contains a list of posts for display, along with an important class, `Post`, which outlines the structure for displayable post information.

**Note that this pull request keeps the logout button for testing purposes.**

| Overview without Posts | Overview with Posts |
|---|---|
| Overview without Posts | Overview with Posts |

### CODE REVIEW

**Comment by CHOOSEIT** [comment]
Following a design review, we decided to update the filter selection.
The new design:

| Overview without Posts | Overview with Posts |
|---|---|
| Overview without Posts | Overview with Posts |

**Comment by gruvw** [comment]
Why was `lib/utils/ui/circular_value.dart` deleted ? Can you please bring it back.

You might not use it in the UI at the moment, but it will be useful when hydrating the UI with the data pulled from the DB.

**Comment by CHOOSEIT** [comment]
I thought that it was maybe a good idea to keep the code as close as what we currently want, and readd later on the necessary utils when needed.

But I can readd it if you want

**Comment by camillelnne** [comment]
Looking great! Very intuitive and simple UI. The code is clean and well-tested.

I am confused by the name of the class `TimeLineFiltersDropDown`. From my understanding, the filters aren't dropdown menus anymore, so it might need to be renamed.

Concerning having the filters stacked on top of each other instead of scrollable, could you please test it on a small phone emulator to ensure that it doesn't take too much space please? Unfortunately my laptop has trouble running it so I can't check.

**Comment by gruvw** [comment]
Add refactoring commits to improve code readability and maintainability, similar comment as https://github.com/ProximaEPFL/proxima/pull/38#issuecomment-2022745226.

Most common issues were:
- mistake methods for local variables
- public facing methods that were supposed to be private
- documentation comment style not correct
- not breaking down widget returned into smaller widgets using local variables in build method

Please @CHOOSEIT take a look at the new code and tell me if there is anything you disagree with or if you have any question about it ?

**Note**: this branch also needs a conflict fix now that #48 as been merged.

**Comment by CHOOSEIT** [comment]
Thank you, @camillelnne, for catching that !

File: lib/views/navigation/routes.dart [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import "package:flutter/material.dart";
 import "package:proxima/views/pages/create_account_page.dart";
-import "package:proxima/views/pages/home_page.dart";
+import "package:proxima/views/pages/home/home_page.dart";
```
Comment (TheTexanCodeur) [inline lib/views/navigation/routes.dart:3]: Good idea to refractor this huge screen

File: lib/views/filter_widgets/timeline_filter/timeline_filters.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/// This enum is used to create the timeline filters.
+/// It contains the name of the filters.
+enum TimeLineFilters {
+  nearest("Nearest"),
+  best("Best"),
+  newFilter("New"),
+  top("Top"),
+  hot("Hot");
```
Comment (TheTexanCodeur) [inline lib/views/filter_widgets/timeline_filter/timeline_filters.dart:8]: Filters are confusing. We should give them clearer names.

File: test/component/utils/mock_data/home/mock_posts.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import "package:proxima/models/ui/post_data.dart";
```
Comment (TheTexanCodeur) [inline test/component/utils/mock_data/home/mock_posts.dart:1]: We should maybe check corner cases posts  (negative numbers, empty names,...)

File: lib/views/pages/home/posts/post_card/post_card.dart [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      padding: const EdgeInsets.all(8),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          VotesWidget(key: postCardVotesKey, votes: post.votes),
+          InkWell(
+            customBorder: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(20),
+            ),
+            //TODO: Implement the logic to navigate to the post
```
Comment (TheTexanCodeur) [inline lib/views/pages/home/posts/post_card/post_card.dart:37]: We already have a navigation action below.

File: lib/views/pages/home/posts/post_card/votes_widget.dart [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+  const VotesWidget({
+    super.key,
+    required this.votes,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final upvote = IconButton(
+      padding: EdgeInsets.zero,
```
Comment (TheTexanCodeur) [inline lib/views/pages/home/posts/post_card/votes_widget.dart:16]: We can remove this line in my opinion

File: lib/views/pages/home/top_bar/home_top_bar.dart [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final title = Text(
+      titleText,
+      style: Theme.of(context).textTheme.headlineMedium,
+    );
+
+    const userAvatar = CircleAvatar(
+      key: profilePictureKey,
+      child: Text("PR"),
```
Comment (TheTexanCodeur) [inline lib/views/pages/home/top_bar/home_top_bar.dart:23]: This needs to be personalized (not a hardcode like here)

File: test/component/home/static_home_test.dart
Comment (TheTexanCodeur) [inline test/component/home/static_home_test.dart:1]: Good test! It covers the displayed elements greatly!

**Review by TheTexanCodeur — CHANGES_REQUESTED** [review]
Great work overall! The code is clean and well structured. Just check the few comments that I wrote and add TODOS if you can't address them right now.

File: test/component/utils/mock_data/home/mock_posts.dart (hunk 3 above)
Comment (CHOOSEIT) [reply test/component/utils/mock_data/home/mock_posts.dart:1]: Yes, but I think that we should that in mind when working on the view model.

File: lib/views/pages/home/posts/post_card/post_card.dart (hunk 4 above)
Comment (CHOOSEIT) [reply lib/views/pages/home/posts/post_card/post_card.dart:37]: Hello, I am not sure to understand why you mean by 

> We already have a navigation action below

In my code, clicking on the comment part of the bottom bar should be done using this `onTap: () => {},` to cover the entire space taken by the widget.

File: lib/views/filter_widgets/timeline_filter/timeline_filters.dart (hunk 2 above)
Comment (CHOOSEIT) [reply lib/views/filter_widgets/timeline_filter/timeline_filters.dart:8]: I changed the filters to sort option, and also changed the `newFilter` instance to `lastest` since the `new` keyword is already reserved by Flutter.

File: lib/views/pages/home/posts/post_card/votes_widget.dart (hunk 5 above)
Comment (CHOOSEIT) [reply lib/views/pages/home/posts/post_card/votes_widget.dart:16]: Hello, it is used to reduce the size of the InkWell and padding created by IconButton widget. See: https://stackoverflow.com/questions/50381157/how-do-i-remove-flutter-iconbutton-big-padding

I added a note in the code thanks.

File: lib/views/pages/home/top_bar/home_top_bar.dart (hunk 6 above)
Comment (CHOOSEIT) [reply lib/views/pages/home/top_bar/home_top_bar.dart:23]: This is a static overview UI, this should be implemented along with the view model.

Comment (CHOOSEIT) [reply lib/views/pages/home/top_bar/home_top_bar.dart:23]: This is currently a static overview UI. I am letting flexibility to the person who will implement the view model.

File: lib/views/pages/home/posts/post_card/post_card.dart (hunk 4 above)
Comment (TheTexanCodeur) [reply lib/views/pages/home/posts/post_card/post_card.dart:37]: The Inkwell widget in the Card already has an ontap function that navigates to the post. I don't understand why the postbottombar should do the same.

**Review by TheTexanCodeur — APPROVED** [review]
Thanks for your clarifications! 

I think that this PR is now ready to be merged with the main. Good job!

File: lib/views/pages/home/posts/post_card/post_card.dart (hunk 4 above)
Comment (CHOOSEIT) [reply lib/views/pages/home/posts/post_card/post_card.dart:37]: Oh! I see ! It was to separate the click on the post from the click on the number of comments. It is purely for the UI 😄 

In terms of UI, it makes an InkWell response on the entire card (post + comment card) when you click on the post. And if you click on the comment number/ icon, you will see a small InkWell response only on the small button. 
😄 

https://github.com/ProximaEPFL/proxima/assets/53620532/5fd1c0a4-aecc-4b41-b897-1f9bfadfc839

**Review by camillelnne — APPROVED** [review]
LGTM!

Great job :)
